concrete/src/User/CsvWriter.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Concrete\Core\User;
+
+use Concrete\Core\Attribute\Category\UserCategory;
+use League\Csv\Writer;
+
+/**
+ * A Writer class for user objects
+ */
+class CsvWriter
+{
+
+    /** @var Writer The writer we use to output */
+    protected $writer;
+
+    /** @var UserCategory */
+    protected $category;
+
+    public function __construct(Writer $writer, UserCategory $category)
+    {
+        $this->writer = $writer;
+        $this->category = $category;
+    }
+
+    /**
+     * Insert a header row for this result set
+     */
+    public function insertHeaders()
+    {
+        $this->writer->insertOne(iterator_to_array($this->getHeaders()));
+    }
+
+    /**
+     * Insert all data from the passed userlist
+     * @param \Concrete\Core\User\UserList $list
+     */
+    public function insertUserList(UserList $list)
+    {
+        $this->writer->insertAll($this->projectList($list));
+    }
+
+    /**
+     * A generator that takes a UserList and converts it to CSV rows
+     * @param \Concrete\Core\User\UserList $list
+     * @return \Generator
+     */
+    private function projectList(UserList $list)
+    {
+        $statement = $list->deliverQueryObject()->execute();
+
+        foreach ($statement as $result) {
+            $user = $list->getResult($result);
+            yield iterator_to_array($this->projectUser($user));
+        }
+    }
+
+    /**
+     * Turn a user into an array
+     * @param \Concrete\Core\User\UserInfo $user
+     * @return array
+     */
+    private function projectUser(UserInfo $user)
+    {
+        yield $user->getUserName();
+        yield $user->getUserEmail();
+
+        $userRegistrationDate = $user->getUserDateAdded();
+        yield $userRegistrationDate->format('d/m/Y G:i');
```

Comment:
I'd avoid calling `format` directly, because the results will be in the system timezone.
I'd use our great date helper, which converts automatically the timezone to the user's one.
Furthermore, I'd use the ISO 8601 date format, to avoid misunderstandings.
For instance, 1/12/2017 is January 12 or December 1? Having 2017-12-01 is much better imho.

So: what about `Core::make('date')->formatCustom('Y-m-d H:i:s')` or `Core::make('date')->formatCustom('Y-m-d\TH:i:s')` ?